Parser.cpp

```diff
@@ -0,0 +1,40 @@
+#include "Parser.h"
+
+const std::map <std::string, std::string> Parser::parseJson(std::string json) {
+    static const std::regex parseRegex("\\s*\"([\\w]*)\"\\s*:\\s*\"?([\\w\\.]*)\"?\\s*[,}]\\s*");
```

Comment:
Értem én, hogy a reguláris kifejezés használata csábító, de ez csak addig működik, amíg a json fájl szerkezete tökéletesen szabályos. Jó lenne felkészíteni ezt az alapvető hibák detektálására, pl. kulcsok hiánya, kulcs vagy érték hiánya a párból, hiányzó vessző, stb., és ezekhez legalább egy általános hibajelzés (kivétel?) küldése.

Reply:
4 hiba detektálására képes a program; ha nincs a fájl elején vagy végén a "{" (vagy a "}") karakter, illetve ha hiányzik egy kulcs vagy egy érték. A hiányzó vesszők esetén az adott sor nem kerül beolvasásra, így az nem is lesz eltárolva.